components/dashboard/src/components/UsageBasedBillingConfig.tsx

```diff
@@ -255,8 +262,15 @@ export default function UsageBasedBillingConfig({ attributionId }: Props) {
                             <div className="flex-grow">
                                 <div className="uppercase text-sm text-gray-400 dark:text-gray-500">Current Period</div>
                                 <div className="text-sm font-medium text-gray-500 dark:text-gray-400">
-                                    <span className="font-semibold">{`${billingPeriodFrom.format("MMMM YYYY")}`}</span>{" "}
-                                    {`(${billingPeriodFrom.format("MMM D")}` + ` - ${billingPeriodTo.format("MMM D")})`}
+                                    <span className="font-semibold">{`${billingCycleFrom.format("MMMM YYYY")}`}</span> (
+                                    <span title={billingCycleFrom.toDate().toUTCString().replace("GMT", "UTC")}>
+                                        {billingCycleFrom.format("MMM D")}
+                                    </span>{" "}
+                                    -{" "}
+                                    <span title={billingCycleTo.toDate().toUTCString().replace("GMT", "UTC")}>
+                                        {billingCycleTo.format("MMM D")}
+                                    </span>
+                                    )
```

Comment:
Note: This is the same code as before -- I've just added helpful `title` attributes so that you can hover to easily see precisely what the billing cycle dates are (including the exact time of the day).